### PR TITLE
Support Plan Preview for Lambda

### LINKED ISSUE #5045: Support Plan Preview for Lambda
This issue includes implementing Diff() func

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

Enabled Plan Preview for Lambda, which compares FunctionManifests.

This is almost the same as ECS's PlanPreview(#4881).

**Which issue(s) this PR fixes**:

Fixes #5045 

**Does this PR introduce a user-facing change?**: no

- **How are users affected by this change**: no
- **Is this breaking change**: no
- **How to migrate (if breaking change)**: no

**How it works:**

When calling PlanPreview as GitHub Actions:
image

### CODE REVIEW

File: pkg/app/piped/platformprovider/lambda/diff.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	var b strings.Builder
+	opts := []diff.RenderOption{
+		diff.WithLeftPadding(1),
+	}
+	renderer := diff.NewRenderer(opts...)
+	if !opt.UseDiffCommand {
+		b.WriteString(renderer.Render(d.Diff.Nodes()))
+	} else {
+		d, err := renderByCommand(diffCommand, d.Old, d.New)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (t-kikuc) [reply pkg/app/piped/platformprovider/lambda/diff.go]: @ffjlabo Thank you!  I fixed it [8d021ab](https://github.com/pipe-cd/pipecd/pull/5046/commits/8d021abf8ccabcee64f3c4b9e3e5dd939fcff525)

**Review by ffjlabo — APPROVED** [review]
LGTM

**Review by khanhtc1202 — APPROVED** [review]
🚀

**Pull request merged**